File: Lab4/MateuszLopacinskiEFProducts/MateuszLopacinskiEFProducts/Invoice.cs
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace MateuszLopacinskiEFProducts
{
    internal class Invoice
    {
        [Key]
        public int InvoiceNumber { get; set; }
        public int Quantity { get; set; }

        public ICollection<Product> Products { get; set; } = new List<Product>();

        public override string ToString()
        {
            StringBuilder sn = new StringBuilder($"Invoice {InvoiceNumber}:");
            foreach (Product product in Products)
            {
                sn.Append($"\t- {product}");
            }
            return sn.ToString();
        }
    }
}

File: Lab5/7/Invoice.java
package com.matipl01;

import javax.persistence.*;
import java.util.Collection;
import java.util.HashSet;

@Entity
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "Invoice_GEN")
    @SequenceGenerator(name = "Invoice_GEN", sequenceName = "Invoice_SEQ")
    private int invoiceNumber;

    private int quantity = 0;

    @ManyToMany(cascade = CascadeType.PERSIST)
    private Collection<Product> products = new HashSet<>();

    public Invoice() {}

    @Override
    public String toString() {
        return String.valueOf(invoiceNumber);
    }

    public void addProduct(Product product, int quantity) {
        this.products.add(product);
        this.quantity += quantity;
    }

    public int getInvoiceNumber() {
        return invoiceNumber;
    }

    public int getQuantity() {
        return quantity;
    }

    public Collection<Product> getProducts() {
        return products;
    }
}
